shithead_game_server/src/lobby.rs

```diff
@@ -233,11 +394,59 @@ impl Lobby {
                 .expect("not enough cards to initialize game")
                 .collect();
         }
+
+        // start the first turn
+        let first_turn_player_id = self.player_list.first_turn_player();
+        self.set_current_turn_and_update_players(first_turn_player_id);
     }
 
     /// Returns the lobby player with the given id.
     pub fn get_player(&self, player_id: ClientId) -> &LobbyPlayer {
-        &self.players[&player_id]
+        &self.player_list[player_id]
+    }
+
+    /// Finds the id of the player who will play in the next turn.
+    fn next_turn_player_id(&self) -> ClientId {
+        match &self.current_turn {
+            Some(current_turn) => self
+                .player_list
+                .find_next_turn_player_after(current_turn.player_id),
+            None => self.player_list.first_turn_player(),
```

Comment:
None wouldn't be matched because the game initiates it from the start, and the calculation for the next index is based on the amount in the game so it will always return an index within the players' IDs. This causes another problem, when the player amount is one then the next index would be: `(0 + 1) % 1` which is still 0. So the player will continue playing against himself.

Reply:
Fixed the first note, and for the second one creates an issue #54